### PR TITLE
Makefile refactor

### PR DESCRIPTION
**Pull request by dtrawins**

### CODE REVIEW

File: Dockerfile.ubuntu [hunk 1]
```diff
@@ -345,13 +348,144 @@ COPY MakefileCapi /ovms/
...
 COPY release_files/LICENSE /ovms/release_files/LICENSE
 COPY client /client
 
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ #
+FROM $BUILD_IMAGE as pkg
+
+RUN mkdir /patchelf 
+WORKDIR /patchelf
+# hadolint ignore=DL3003
+RUN wget -q https://github.com/NixOS/patchelf/archive/0.10.tar.gz && \
```
Comment (jasiu86) [inline Dockerfile.ubuntu:365]: Can we use available deb package in version:

patchelf/focal 0.10-2build1 amd64  (ubuntu 20.04)

File: Makefile [hunk 2]
```diff
@@ -263,86 +297,58 @@ endif
...
-		--build-arg JOBS=$(JOBS)
-
-targz_package: ovms_builder_image
-	docker build $(NO_CACHE_OPTION) -f DockerfileMakePackage . \
-		--build-arg http_proxy=$(HTTP_PROXY) \
-		--build-arg https_proxy="$(HTTPS_PROXY)" \
-		--build-arg ov_use_binary=$(OV_USE_BINARY) \
-		--build-arg sentencepiece=$(SENTENCEPIECE) \
-		--build-arg BASE_OS=$(BASE_OS) \
+		--build-arg JOBS=$(JOBS) \
```
Comment (jasiu86) [inline Makefile:309]: $(JOBS) are already present in $(BUILD_ARGS)  

(line #180)

File: create_package.sh [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+sha256sum ovms.tar.xz > ovms.tar.xz.sha256
+cd /ovms_release
+ls -l
+
+
+
+
+
+
+
```
Comment (jasiu86) [inline create_package.sh:98]: Why so many new lines?

File: create_package.sh [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# This script should be used inside the build image to create a binary package based on the compiled artefacts
+
+
+mkdir -vp /ovms_release/bin && \
```
Comment (jasiu86) [inline create_package.sh:20]: Why not create it in single mkdir call?

File: create_package.sh [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+mkdir -vp /ovms_release/deps && \
+mkdir -vp /ovms_release/lib && \
+mkdir -vp /ovms_release/lib/hddl/config && \
+mkdir -vp /ovms_release/lib/custom_nodes
```
Comment (jasiu86) [inline create_package.sh:24]:
```suggestion
mkdir -vp /ovms_release/bin /ovms_release/deps   /ovms_release/lib /ovms_release/lib/hddl/config /ovms_release/lib/custom_nodes
```

**Review by jasiu86 — APPROVED** [review]
Looks good

File: Dockerfile.redhat [hunk 6]
```diff
@@ -101,16 +105,14 @@ WORKDIR /ovms/third_party/opencv
...
+FROM base_build as build
+ARG BASE_IMAGE
 LABEL version="1.0.0"
 
 SHELL ["/bin/bash", "-xo", "pipefail", "-c"]
 
-ARG ov_source_branch=releases/2021/2
-
-ARG JOBS
+ARG JOBS=40
```
Comment (dkalinowski) [inline Dockerfile.redhat:115]: Why default is 40?

File: Dockerfile.redhat [hunk 7]
```diff
@@ -312,6 +314,7 @@ RUN bash -c "sed -i -e 's|REPLACE_PROJECT_NAME|${PROJECT_NAME}|g' /ovms/src/vers
 RUN if [ "$build_type" = "dbg" ] ; then bash -c "sed -i -e 's|REPLACE_PROJECT_VERSION|${PROJECT_VERSION}-debug|g' /ovms/src/version.hpp" ; else bash -c "sed -i -e 's|REPLACE_PROJECT_VERSION|${PROJECT_VERSION}|g' /ovms/src/version.hpp" ; fi ;
 RUN if [ "$ov_use_binary" = "1" ] ; then true ; else exit 0 ; fi ; sed -i -e "s#REPLACE_OPENVINO_NAME#$(find /opt/intel/ -maxdepth 1 -mindepth 1 -type d | grep openvino | grep -Eo '[0-9]{4}.[0-9].[0-9].[0-9]+.[^_]+')#g" /ovms/src/version.hpp
 RUN if [ "$ov_use_binary" == "0" ] ; then true ; else exit 0 ; fi ; sed -i -e "s#REPLACE_OPENVINO_NAME#$(git --git-dir /openvino/.git log -n 1 | head -n 1 | cut -d' ' -f2 | head -c 12)#g" /ovms/src/version.hpp
+RUN bash -c "sed -i -e 's|REPLACE_BUILD_PARAMS|${debug_bazel_flags}|g' /ovms/src/version.hpp"
```
Comment (dkalinowski) [inline Dockerfile.redhat:317]: `REPLACE_BUILD_PARAMS` may be misleading - build params of what - entire building process? (makefile) image building process? (dockerfile)
In fact this is none of those - params you specify here are params for bazel build step.
My suggestion is to change `BUILD_PARAMS` everywhere to `BAZEL_BUILD_FLAGS`

File: Makefile [hunk 8]
```diff
@@ -138,20 +142,52 @@ ifeq ($(NVIDIA),1)
...
+	--build-arg ov_use_binary=$(OV_USE_BINARY)\
+	--build-arg sentencepiece=$(SENTENCEPIECE)\
+	--build-arg DLDT_PACKAGE_URL=$(DLDT_PACKAGE_URL)\
+	--build-arg CHECK_COVERAGE=$(CHECK_COVERAGE)\
+	--build-arg RUN_TESTS=$(RUN_TESTS)\
+	--build-arg FUZZER_BUILD=$(FUZZER_BUILD)\
+	--build-arg build_type=$(BAZEL_BUILD_TYPE)\
+	--build-arg debug_bazel_flags=$(BAZEL_DEBUG_FLAGS)\
+	--build-arg CMAKE_BUILD_TYPE=$(CMAKE_BUILD_TYPE)\
+	--build-arg minitrace_flags=$(MINITRACE_FLAGS)\
```
Comment (dkalinowski) [inline Makefile:174]: I dont see it being used in any dockerfile. Are you sure minitrace still works? There is test for that in ovms-test.

File: Dockerfile.redhat [hunk 9]
```diff
@@ -158,21 +160,21 @@ RUN if [ "$NVIDIA" == "1" ] ; then true ; else exit 0 ; fi ; \
...
     mkdir -p ccache-4.3/build && cd ccache-4.3/build && \
-    cmake -DCMAKE_BUILD_TYPE=Release -G Ninja .. && \
+    cmake -DCMAKE_BUILD_TYPE=Release -DZSTD_FROM_INTERNET=ON -G Ninja .. && \
     ninja -v install && \
     rm -rf /var/cache/yum
 
 # build_type=[ opt, dbg ]
-ARG build_type=dbg
+ARG build_type=opt
 ARG minitrace_flags
```
Comment (dkalinowski) [inline Dockerfile.redhat:169]: This is unused as far I see

**Review by dkalinowski — APPROVED** [review]

**Pull request merged**